core/src/main/java/org/apache/calcite/rel/RelDistributions.java

```diff
@@ -137,7 +137,7 @@ public RelDistribution apply(Mappings.TargetMapping mapping) {
       return getTraitDef().canonize(
           new RelDistributionImpl(type,
               ImmutableIntList.copyOf(
-                  Mappings.apply((Mapping) mapping, keys))));
```

Comment:
why is this change needed? `apply` and `apply2` seems to be different :
* `apply` reorders `list` elements according to `mapping`
** `outList[k]=inList[mappingS[k]]`
* while `apply2` returns the mapping target(s) according to the list
** `outList[k]=mappingT[inList[k]]`

I would feel that in this case `apply` is the right one to use here - why it's not?

Reply:
I agree: at this location we should have `apply2` and not `apply` - it was a bug that `apply` was used here earlier